extmod/modssl_mbedtls.c

```diff
@@ -236,6 +264,60 @@ STATIC mp_obj_t ssl_context___del__(mp_obj_t self_in) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(ssl_context___del___obj, ssl_context___del__);
 #endif
 
+// Ciphersuites
+
+STATIC mp_obj_t ssl_context_get_ciphers(mp_obj_t self_in) {
+    mp_obj_t list = mp_obj_new_list(0, NULL);
+    for (const int *cipher_list = mbedtls_ssl_list_ciphersuites(); *cipher_list; ++cipher_list) {
+        const char *cipher_name = mbedtls_ssl_get_ciphersuite_name(*cipher_list);
+        mp_obj_list_append(list,
+            MP_OBJ_FROM_PTR(mp_obj_new_str(cipher_name,
+                strlen(cipher_name))));
+        cipher_list++;
+        if (!*cipher_list) {
+            break;
+        }
+    }
+    return list;
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(ssl_context_get_ciphers_obj, ssl_context_get_ciphers);
+
+STATIC mp_obj_t ssl_context_set_ciphers(mp_obj_t self_in, mp_obj_t ciphersuite) {
+    // check that ciphersuite is a list or tuple
+    size_t len = 0;
+    mp_obj_t *ciphers;
+    mp_obj_get_array(ciphersuite, &len, &ciphers);
+    if (len > 0) {
+        mp_obj_ssl_context_t *ssl_context = MP_OBJ_TO_PTR(self_in);
+        ssl_context->ciphersuites = m_new(int, len + 1);
+        for (int i = 0, n = len; i < n; i++) {
+
+            if (ciphers[i] != mp_const_none) {
+                const char *ciphername = mp_obj_str_get_str(ciphers[i]);
+                const int id = mbedtls_ssl_get_ciphersuite_id(ciphername);
+                ssl_context->ciphersuites[i] = id;
+                if (id == 0) {
+                    mbedtls_raise_error(MBEDTLS_ERR_SSL_BAD_CONFIG);
+                }
+            }
+        }
+        ssl_context->ciphersuites[len + 1] = 0;
+
+        // Ciphersuite Config
+        if (ssl_context->ciphersuites[0] != 0) {
```

Comment:
This will always be true, because of the `if (len > 0)` check above, so you can remove this if.